input/dir/dir_input_test.go

```diff
@@ -0,0 +1,103 @@
+package dir
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+	"github.com/spacemonkeygo/errors"
+	"github.com/spacemonkeygo/errors/try"
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/lib/fshash"
+	"polydawn.net/repeatr/testutil"
+)
+
+func Test(t *testing.T) {
+	Convey("Given a nonexistant path", t, func() {
+		Convey("The input config should be rejected during validation", func() {
+			correctError := false
+			try.Do(func() {
+				New(def.Input{
+					Type:     "dir",
+					Hash:     "abcd",
+					URL:      "/tmp/certainly/should/not/exist",
+					Location: "/data",
+					// REVIEW: two of these four fields were more for the executor.
+					// maybe this struct isn't the sanest to use for this arg.
```

Comment:
/shrug, maybe later?  Reconsidering it now by the light of day, it seems kinda silly to worry about too many fields being used in a code path that's basically only ever going to be hit directly by tests.  I'll remove the comment for now and we can reconsider if it still seems awkward in the bigger picture later.